File: models/match.py
from dataclasses import dataclass
from typing import Optional
from datetime import datetime

@dataclass
class Match:
    match_code: str
    league_code: str
    pool_id: int
    team_id_a: int
    team_id_b: int
    match_date: datetime
    status: str
    season: str
    id: Optional[int] = None
    set: Optional[str] = None
    score: Optional[str] = None
    venue: Optional[str] = None
    first_referee: Optional[str] = None
    second_referee: Optional[str] = None
    live_code: Optional[int] = None
    active: bool = True
    created_at: Optional[datetime] = None
    last_update: Optional[datetime] = None

File: types/Match.ts
export enum MatchStatus {
    UPCOMING = "UPCOMING",
    FINISHED = "FINISHED",
}

export interface Match {
    id: number;
    match_code: string;
    league_code: string;
    pool_id: number;
    team_id_a: number;
    team_id_b: number;
    match_date: Date;
    status: MatchStatus;
    set?: string;
    score?: string;
    venue?: string;
    referee1?: string;
    referee2?: string;
    live_code?: number;
    last_update?: Date;
    active: boolean;
}